### PR TITLE
[bitnami/keycloak] Fix Kubernetes discovery, add httpRelativePath to make migration easier

### LINKED ISSUE #10667: Unable to login to high available keycloak cluster
### Name and Version

Helm repo: https://charts.bitnami.com/bitnami -> bitnami/keycloak, Chart version: 5.2.8 , Image version: 15.1.1-debian-10-r10,

### What steps will reproduce the bug?

After installing the helm chart with below values mentioned, 

### Are you using any custom parameters or values?

```
global:
  image:
    registry: docker.io
    repository: bitnami/keycloak
    tag: 15.1.1-debian-10-r10
    pullPolicy: IfNotPresent
    pullSecrets: []
    debug: true

proxyAddressForwarding: true

serviceDiscovery:
  enabled: true
  protocol: dns.DNS_PING
  properties:
    - dns_query=keycloak.identity.svc.cluster.local  
  transportStack: tcp

cache:
  ownersCount: 3
  authOwnersCount: 3

replicaCount: 3

ingress:
  enabled: true
  hostname: my-keycloak.keycloak.example.com
  apiVersion: ""
  ingressClassName: "nginx"
  path: /
  pathType: ImplementationSpecific
  annotations: {}
  tls: false
  extraHosts: []
  extraTls: []
  secrets: []
  existingSecret: ""
  servicePort: http  
```

### What is the expected behavior?

User successfully login to keycloak UI

### What do you see instead?

0:07:05,251 WARN  [org.keycloak.events] (default task-1) type=CODE_TO_TOKEN_ERROR, realmId=master, clientId=security-admin-console, userId=null, ipAddress=10.244.5.46, error=invalid_code, grant_type=authorization_code, code_id=157e0483-67fa-4ea4-a964-387f3884cbc9, client_auth_method=client-secret

### Additional information

_No response_

### LINKED ISSUE #10885: [bitnami/keycloak] Readiness/Livenessprobe fails when Keycloak is served from subpath only
### Name and Version

bitnami/keycloak 9.3.2

### What steps will reproduce the bug?

My infrastructure needs to reach Keycloak at /auth only which was somewhat deprecated with Keycloak Quarkus. Thanks to a recent pull request I can use the following config to access Keycloak just like before:

```yaml
proxy: edge
extraEnvVars:
  - name: KEYCLOAK_HTTP_RELATIVE_PATH
    value: auth
  - name: KC_HOSTNAME
    value: my-domain
## Ingress configuration
##
ingress:
  enabled: true
  ingressClassName: nginx
  hostname: my-domain
  path: /auth
```

### Are you using any custom parameters or values?

_No response_

### What is the expected behavior?

_No response_

### What do you see instead?

The problem is [Keycloak's readiness probe in the statefulset](https://github.com/bitnami/charts/blob/master/bitnami/keycloak/templates/statefulset.yaml#L311) looks at /realms/master, which can't be reached and should be /auth/realms/master. 

> Warning  Unhealthy  30s (x2 over 40s)  kubelet            Readiness probe failed: Get "http://192.168.173.115:8080/realms/master": dial tcp 192.168.173.115:8080: connect: connection refused
>   Warning  Unhealthy  0s (x3 over 20s)   kubelet            Readiness probe failed: HTTP probe failed with statuscode: 404

This was changed with the introduction of Keycloak 17. [The commit](https://github.com/bitnami/charts/pull/10095/files#diff-13d578ae617ed76deafb93a8bdcf0e71e8a279e46a1ea9c99fc7e071ba3e0a00)

### Additional information

When I template the chart to a YAML file and manually set the path of the probes the checks don't fail.

```yaml
readinessProbe:
  failureThreshold: 3
  httpGet:
    path: /auth/realms/master
    port: http
livenessProbe:
  failureThreshold: 3
  httpGet:
    path: /auth
    port: http
```

Maybe it could be possible to add .`Values.ingress.path` as prefix to the probes?

### PR DESCRIPTION
**Pull request by captnbp**
### Description of the change

- Fix `-Djgroups.dns.query` headless service used for discovery by using FQDN (cf. https://www.keycloak.org/server/caching#_transport_stacks)
- Use Keycloak image with fixed cache-stack settings
- Add  httpRelativePath to make migration from older Keycloak version easier as they used `/auth/` as relative path

### Benefits

- Fix cluster discovery with ISPN cache
- Use Keycloak image with fixed cache-stack settings
- Help users to migrate from older Keycloak version to 18.x

### Possible drawbacks

None

### Applicable issues

  - fixes #10667
  - fixes #10885
 
### Additional information

None

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by FraPazGal** [comment]
Hello @captnbp, thanks for the PR! The changes LGTM but could you please merge the latest changes from master? We did some renaming to our GH Actions and a merge is required so the new actions are run. Additionally, could you please resolve the conflict on the README?

**Comment by captnbp** [comment]
@FraPazGal I rebase everything and regenerated the README.md file.
Hope it will be reviewed / merged before any other MR that would break again my MR :-/

**Review by FraPazGal — DISMISSED** [review]
LGTM, thanks for the PR! Let's wait until @alemorcuq gives the final approval before merging.

**Review by alemorcuq — APPROVED** [review]

**Pull request merged**
